### ISSUE TITLE
Regarding your blogpost

### ISSUE THREAD

**Issue by nps1ngh** [mhmoodlan/b-cos-robustness#1]
Hey,

Thanks a lot for your interest in [B-cos/B-cos-v2](https://github.com/B-cos/B-cos-v2) and your merge request!

Also thanks for taking a look the robustness of B-cos networks :smile:. This is something that we've had on our radar for a while, but just couldn't find the time for it. I read your [blog post](https://mhmoodlan.github.io/blog/b-cos-robustness) and really liked it, especially the visualizations!

However, I have a few important notes to make:

1. Comparing a ResNet-50 to a ResNet-56 might not be a fair comparison, as a R50 contains about 23.5M parameters and a R56 only about 852K parameters. In fact, I feel like a R50 might be a little overkill for C10, but for a fair comparison, I would suggest training a B-cos R50 for a fairer comparison.

2. This is more of naming mistake on my part: the `resnet_56-nomaxout` config actually doesn't have any norm layers, hence the low accuracy (also no hyper-parameter tuning was done, we just took the hyper-parameters for a simple B-cos CNN). The R50 contains norm layers, and hence for a fairer comparison, I suggest using a model with norm layers as well (the B-cos R50 configs have no maxout and come with norm layers, you would just need to change the number of class outputs). Additionally for reference: the `resnet_56_bnu-nomaxout` model has a norm layer and no maxout, and it should achieve about `91.3%` accuracy.

3. The last point brings me to another thing: we actually did not report any accuracy numbers for a B-cos R56 in the paper (there just wasn't enough space, we were already a little beyond the limit). The numbers reported there are from a v1 model (you can find the configs [here](https://github.com/moboehle/B-cos/blob/main/experiments/CIFAR10/bcos/experiment_parameters.py)), which has been tuned a little. In the paper, we only mention that it's comparable to a standard R56 from the original ResNet paper.

Summing up: I think to facilitate a fairer comparison to the R50 (23.5M parameters) from the `robustness` library from MadryLabs, we should compare that to a similar B-cos R50 and not a B-cos R56 (852K parameters). (It'd be great to use similar hyper-parameters, only slightly changed to make them more compatible with B-cos (if you have any questions regarding this, I'd be happy to help :smile:).)

(Also it might be interesting to qualitatively evaluate how the explanations change for the ImageNet models.)

Again thanks a lot for your interesting work!

**Comment by mhmoodlan** [comment]
Hey,

Thank you for the thoughtful notes! 
- Regarding fair comparison: I totally agree with you, I should've made it clear in the post, by no means this is a fair evaluation of robustness. When I came across the CoDA and B-cos papers I simply got too excited 😄 and wanted to have a rough idea on their robustness. Also, I was constrained by google colab's free compute budget, which prevented any serious training of larger models (without losing brain cells).
- About B-cos R56 accuracy: sorry for misquoting your paper, I made proper adjustments to the post.

Once I get my hands on some GPUs, I will try to run the B-cos R50 experiments you suggested. Thanks!
